metis-indexing/metis-indexing-service/src/main/java/eu/europeana/indexing/service/PublishingService.java

```diff
@@ -0,0 +1,404 @@
+    	}
+
+    }
+    
+    @Autowired
+    public PublishingService(FullBeanDao fullBeanDao, LBHttpSolrServer solrServer, CloudSolrServer cloudSolrServer) {
+  	  this.fullBeanDao = fullBeanDao;
+  	  this.solrServer = solrServer;
+  	  this.cloudSolrServer = cloudSolrServer;
+    }
+    
+    public boolean process(String record) throws JiBXException, SolrServerException
+    {    	
+    	RDF rdf = toRDF(record); 
+    	
+    	FullBeanImpl fBean = null;
+		try {
+			fBean = new MongoConstructor().constructFullBean(rdf);
+		} catch (InstantiationException | IllegalAccessException | IOException e) {
+			e.printStackTrace();
+		}
+    	      	
+    	if (fBean != null) {
+    		fBean.setEuropeanaCollectionName(new String[100]); // To prevent null pointer exception using sample
+    		    		
+    		SolrDocumentHandler solrDocHandler = new SolrDocumentHandler(cloudSolrServer);
+    		SolrInputDocument solrInputDoc = solrDocHandler.generate(fBean);
+    		
+    		try {
+				cloudSolrServer.add(solrInputDoc);
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+    		    		  
+            FullBeanImpl saved;
+            
+            if (fBean.getAbout() == null) {
+                try {
+					saveEdmClasses(fBean, true, fullBeanDao);
+				} catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+                
+                fullBeanDao.save(fBean);
+                
+                saved = fullBeanDao.getFullBean(fBean.getAbout());
+            } else {
+                try {
+					saved = updateFullBean(fBean, fullBeanDao);
+				} catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+            }            
+            
+            return true;
+    	}
+
+    	return false;
+    }
+    
+    public static RDF toRDF(String xml) throws JiBXException {
+    	IUnmarshallingContext context = getRdfBindingFactory().createUnmarshallingContext();
+    	return (RDF) context.unmarshalDocument(IOUtils.toInputStream(xml), UTF8);
+    }
+
+    private static IBindingFactory getRdfBindingFactory() {
+    	if (rdfBindingFactory != null) {
+    		return rdfBindingFactory;
+    	}
+    	throw new IllegalStateException("No binding factory available.");
+    }
+
+    public static String convertRDFtoString(RDF rdf) throws JiBXException, UnsupportedEncodingException {
+    	IMarshallingContext context = getRdfBindingFactory().createMarshallingContext();
+    	context.setIndent(2);
+    	ByteArrayOutputStream out  = new ByteArrayOutputStream();
+    	context.marshalDocument(rdf, UTF8, null, out);
+    	return out.toString(UTF8);
+    }
+    
+    private static void saveEdmClasses(FullBeanImpl fullBean, boolean isFirstSave, FullBeanDao fullBeanDao) throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {

```

Comment:
This method has a Cyclomatic complexity of 30 out of 15 allowed. It's ready huge and should be split into more constructed sub methods.